connector/connect/common/src/main/protobuf/spark/connect/base.proto

```diff
@@ -158,6 +159,9 @@ message ExecutePlanResponse {
   // batch of results and then represent the overall state of the query execution.
   Metrics metrics = 4;
 
+  // The metrics observed during the execution of the query plan.
+  optional ObservedMetrics observed_metrics = 5;
```

Comment:
I think this should be `repeated`?  we may have multi observations in one plan

```
In [10]: observation1 = Observation("my metrics 1")

In [11]: observation2 = Observation("my metrics 2")

In [12]: observed_df = df.observe(observation1, count(lit(1)).alias("count"), max(col("age"))).withColumn("xyz", lit(1)).observe(observation
    ...: 2, count(lit(1)).alias("count"), max(col("name")))

In [13]: observed_df.count()
Out[13]: 2

In [14]: observation1.get
Out[14]: {'count': 2, 'max(age)': 5}

In [15]: observation2.get
Out[15]: {'count': 2, 'max(name)': 'Bob'}

In [16]: observed_df.explain()
== Physical Plan ==
CollectMetrics my metrics 2, [count(1) AS count#42L, max(name#1) AS max(name)#44]
+- *(2) Project [age#0L, name#1, 1 AS xyz#37]
   +- CollectMetrics my metrics 1, [count(1) AS count#34L, max(age#0L) AS max(age)#36L]
      +- *(1) Scan ExistingRDD[age#0L,name#1]
```

Reply:
may bad, `ObservedMetrics` can contains multi observations

Reply:
Yes. but let `ObservedMetricsObject` be `repeated`.